packages/api/src/jobs/projects-cleanup.test.ts

```diff
@@ -0,0 +1,129 @@
+import { v4 as uuid } from "uuid";
+
+import * as appRouter from "../app-router";
+import * as projectsController from "../controllers/project";
+import { cache } from "../store/cache";
+import { DB } from "../store/db";
+import Queue, { RabbitQueue } from "../store/queue";
+import { rabbitMgmt } from "../test-helpers";
+import params, { testId } from "../test-params";
+import projectsCleanup from "./projects-cleanup";
+import { Request } from "express";
+
+describe("projects-cleanup", () => {
+  // There are further functional tests under controllers/stream.test.ts "active clean-up"
+
+  let db: DB;
+  let initClientsSpy: jest.SpyInstance;
+
+  beforeAll(async () => {
+    db = new DB();
+    await db.start({ postgresUrl: params.postgresUrl });
+    await rabbitMgmt.createVhost(testId);
+  });
+
+  afterAll(async () => {
+    await rabbitMgmt.deleteVhost(testId);
+  });
+
+  let queue: Queue;
+
+  beforeEach(() => {
+    const originalInitClient = appRouter.initClients;
+    initClientsSpy = jest
+      .spyOn(appRouter, "initClients")
+      .mockImplementation(async (params, name) => {
+        const result = await originalInitClient(params, name);
+        queue = result.queue;
+        jest.spyOn(queue, "consume");
+        jest.spyOn(queue, "delayedPublishWebhook");
+        return result;
+      });
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+    cache.storage = null;
+    queue?.close();
+    queue = null;
+  });
+
+  const mockProject = (deleted: boolean, deletedAt: number) => {
+    return {
+      id: uuid(),
+      name: "project1",
+      deleted,
+      deletedAt,
+    };
+  };
+
+  const mockStream = (projectId: string) => {
+    return {
+      id: uuid(),
+      name: "stream1",
+      playbackId: uuid(),
+      streamKey: uuid(),
+      projectId,
+    };
+  };
+
+  const mockAsset = (projectId: string) => {
+    return {
+      id: uuid(),
+      name: "asset1",
+      projectId,
+      source: {
+        type: "url" as const,
+        url: "someSource",
+      },
+    };
+  };
+
+  it("it should deleted related assets and streams", async () => {
+    const triggerSpy = jest
+      .spyOn(projectsController, "triggerCleanUpProjectsJob")
+      .mockImplementation(() => [[], Promise.resolve()]);
```

Comment:
Same regarding the "inverted" nature of the current jobs implementation. Let's make sure the "real" tests are implemented here, which check the effects of the deletion job. The API that should be the "side-kick" which calls the job, but the core logic of the job should be the job itself